qtdialogs/DlgSetupManager.py

```diff
@@ -569,127 +491,80 @@ def _addActionButton(self, item, text, handler):
       """Create and add a left-aligned action button to the wallet list item."""
       btn = QtWidgets.QPushButton(text)
       btn.clicked.connect(handler)
-      btn.setMaximumWidth(60)  # Compact button
+      btn.setMaximumWidth(60)
       cellWidget = QtWidgets.QWidget()
       layout = QtWidgets.QHBoxLayout(cellWidget)
-      layout.setContentsMargins(2, 2, 2, 2)  # Small margins
+      layout.setContentsMargins(2, 2, 2, 2)
       layout.addWidget(btn)
-      layout.addStretch()  # Push button to left
+      layout.addStretch()
       self.walletList.setItemWidget(item, 4, cellWidget)
 
-   def _createWalletRow(self, entry, stagedWalletIds=None):
-      """Create and add a wallet row from a bridge entry.
-
-      Returns True if the row is rendered successfully; False otherwise.
-      """
+   def _renderWalletRow(self, wallet_info):
+      """Render a single wallet row from prepared wallet data (UI ONLY)."""
       try:
-         walletPath = entry.path
-         fileName = os.path.basename(walletPath) if walletPath else ''
-         walletId = entry.walletId
-
-         # Check if this is a legacy wallet that has been migrated
-         # Per BridgeTests.cpp: after migration, there are 2 wallets with
-         # same walletId:
-         # - Legacy: staged=false (original file) <- Hide this one
-         # - Migrated: staged=true (new file) <- Show this one
-         isStaged = entry.staged or False
-
-         # Hide legacy wallets ONLY if there's a staged version
-         # Use only staged status since loadState doesn't exist in capnp
-         if not isStaged and stagedWalletIds:
-            # Check if there's a staged wallet with the same walletId
-            # (staged version)
-            if walletId in stagedWalletIds:
-               return False  # Skip the legacy version
-         # Create row
+         wallet_id = wallet_info['wallet_id']
```

Comment:
Please, no. Most of these values are referred to only once, don't spray out at the top of the method whole sale. At any rate, use a dedicated class, not a dict.